From 13c96c5b797fd7f92a657af984626115c45212e3 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:52:58 -0400
Subject: [tools] perf: Ensure sscanf does not overrun the "mem" field

Message-id: <1408953210-26343-291-git-send-email-jolsa@redhat.com>
Patchwork-id: 89027
O-Subject: [PATCH RHEL7.1 BZ1133083 290/322] perf tools: Ensure sscanf does not overrun the "mem" field
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit a761a2d8a7175b7b4e8525e0672e1a8d3c051001
Author: Alan Cox <alan@linux.intel.com>
Date: Mon Jan 20 19:10:11 2014 +0100

description
===========
Make the parsing robust.

(perf has some other assumptions that BUFSIZE <= MAX_PATH which are
 not touched here)

Reported-by: Jackie Chang
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/header.c b/tools/perf/util/header.c
index bb3e0ed..893f8e2 100644
--- a/tools/perf/util/header.c
+++ b/tools/perf/util/header.c
@@ -930,7 +930,7 @@ static int write_topo_node(int fd, int node)
   /* skip over invalid lines */
   if (!strchr(buf, ':'))
    continue;
-  if (sscanf(buf, "%*s %*d %s %"PRIu64, field, &mem) != 2)
+  if (sscanf(buf, "%*s %*d %31s %"PRIu64, field, &mem) != 2)
    goto done;
   if (!strcmp(field, "MemTotal:"))
    mem_total = mem;
-- 
1.7.1